### PR TITLE
Network Observability 1.5 release notes

### PR DESCRIPTION
**Pull request by skrthomas**
Merge to only the no-1.5 branch - no cherrypicks are required.
This PR is part of an experiment for simplifying merges for asynchronous content, and I will open one PR against main to incorporate all of the Network Observability 1.5 content just before its GA.
Version(s):

4.12+
Issue:

## New features and enhancements
- Zones: https://issues.redhat.com/browse/OSDOCS-9419
- Round-trip time: https://issues.redhat.com/browse/OSDOCS-7593
- Improved LokiStack integration: https://issues.redhat.com/browse/OSDOCS-8253
- DNS enhancements: https://issues.redhat.com/browse/OSDOCS-8429
- Dashboard enhancements for Lokiless use: https://issues.redhat.com/browse/OSDOCS-8427
- Update to Network Traffic overview (under Web console enhancements in the RN): https://issues.redhat.com/browse/OSDOCS-8465
- `kafkaBatchSize` update under Notable enhancements-> Performance enhancements: https://issues.redhat.com/browse/NETOBSERV-1470

## Bug fixes
- https://issues.redhat.com/browse/NETOBSERV-1134
- https://issues.redhat.com/browse/NETOBSERV-1351
- https://issues.redhat.com/browse/NETOBSERV-1430
- https://issues.redhat.com/browse/NETOBSERV-1464
- https://issues.redhat.com/browse/NETOBSERV-1443
- https://issues.redhat.com/browse/NETOBSERV-1305
- https://issues.redhat.com/browse/NETOBSERV-1293
## Known Issues
- NetObserv plugin not showing: https://issues.redhat.com/browse/NETOBSERV-1493

Link to docs preview:
https://70738--docspreview.netlify.app/openshift-enterprise/latest/network_observability/network-observability-operator-release-notes#network-observability-operator-release-notes-1-5

QE review:
- [X] QE has approved this change.

Additional information:

*NOTE*: 1.4.2 release notes were not available at the time the `no-1.5` branch for documentation was created; therefore they are not included in this preview. When `no-1.5` docs branch is merged with `main`, I will rebase the release notes and the 1.4.2 note will be added in.

### CODE REVIEW

File: network_observability/network-observability-operator-release-notes.adoc
Comment (jpinsonneau) [inline network_observability/network-observability-operator-release-notes.adoc:49]: PR is still pending, let's keep an eye on it but I feel this is not going to be part of 1.5:
https://issues.redhat.com/browse/NETOBSERV-1382
https://github.com/netobserv/network-observability-console-plugin/pull/431

File: network_observability/network-observability-operator-release-notes.adoc
Comment (jpinsonneau) [inline network_observability/network-observability-operator-release-notes.adoc:56]:
```suggestion
The 1.5 release of the Network Observability Operator adds improvements and new capabilities to the {product-title} web console plugin and the Operator configuration.
```

File: network_observability/observing-network-traffic.adoc
Comment (skrthomas) [inline network_observability/observing-network-traffic.adoc:34]: This was accidentally removed in a rebase of this[PR](https://github.com/openshift/openshift-docs/pull/69106/files#diff-21cba4b9eea6286345ee8e2ad7b3cc0e4dc6ac125a47623a7df9f52f2f2506aaL27-L31)

File: network_observability/network-observability-operator-release-notes.adoc
Comment (jpinsonneau) [inline network_observability/network-observability-operator-release-notes.adoc:55]: There are actually a bunch of UI improvements here. Do we want to list them all ?

See:
https://issues.redhat.com/browse/NETOBSERV-657
https://issues.redhat.com/browse/NETOBSERV-676

Such as:
- Collection latency warning in time range selection popup
- Panels & columns popups quick filters
- New graphs available for DNS & RTT: Min, Max, P90, P99
- Switch graph type
- Show Top / Overall

Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:55]: Yeah I guess we don't necessarily document all of these improvements, but I agree we can point them all out.

File: network_observability/observing-network-traffic.adoc
Comment (skrthomas) [reply network_observability/observing-network-traffic.adoc]: I merged another PR to add this back in.

**Comment by kalexand-rh** [comment]
4.11 is now [EOL](https://access.redhat.com/support/policy/updates/openshift), and we are no longer updating it. Please set the versions to 4.12+.

**Comment by nathan-weinberg** [comment]
LGTM, @memodi will give final approval from QE team

File: network_observability/network-observability-operator-release-notes.adoc
Comment (memodi) [inline network_observability/network-observability-operator-release-notes.adoc:53]:
```suggestion
* New panel display option to focus on one panel while keeping others viewable but smaller focus, switch graph type, and show Top / Overall.
```

File: network_observability/network-observability-operator-release-notes.adoc
Comment (memodi) [inline network_observability/network-observability-operator-release-notes.adoc:85]: @skrthomas : should we add Performance bug as a known issue? cc @openshift/netobserv-team

**Review by memodi — COMMENTED** [review]
One typo and a question on performance bug.

I noticed we don't mention DSCP feature: https://issues.redhat.com/browse/NETOBSERV-1335, is that intentional?

**Review by OlivierCazade — APPROVED** [review]
LGTM

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:85]: @memodi I think in the past releases, we've decided to leave general performance bugs out of the known issues. The only time I think we noted it was when there was a specific configuration that was related to performance. That being said, I defer to SMEs about this.

Comment (memodi) [reply network_observability/network-observability-operator-release-notes.adoc:85]: @skrthomas - no worries. We had discussion in QE sync, I am fine with it being left out of RN and we can keep working it internally.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [inline network_observability/network-observability-operator-release-notes.adoc:56]: @memodi as discussed on Slack, I added this bullet back. DSCP is only a release note mention. We do not have any other product documentation for using DSCP, as it is my understanding based on conversation with Amogh that DSCP is not something that needs to be configured by Network Observability customers. Its just a field that is now available by default for filtering in the Network Traffic table, which we don't document the complete list of those fields in the Network Traffic table. Please let me know if I'm wrong about any of this 😅

Comment (memodi) [reply network_observability/network-observability-operator-release-notes.adoc:56]: that's right. No additional configuration is required in flowcollector on customers side. Thanks.

**Review by jpinsonneau — APPROVED** [review]

File: network_observability/network-observability-operator-release-notes.adoc
Comment (michaelryanpeter) [inline network_observability/network-observability-operator-release-notes.adoc:72]:
```suggestion
* Previously, the *NetObserv/Health* dashboard displayed an empty graph named *Flows Overhead* if you used the default metrics setting. The *Flows Overhead* metric was still available if you removed `namespaces-flows` and `namespaces` from the list of ignored tags. Now, using the default metrics settings, this metric is visible. (link:https://issues.redhat.com/browse/NETOBSERV-1351[*NETOBSERV-1351*])
```

I am not sure about how to label the graph/metric. It might be worth looking at the ISG for guidance.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (michaelryanpeter) [inline network_observability/network-observability-operator-release-notes.adoc:85]: I think these bug fixes and known issues need another pass for style and grammar consistency. Right now, there are several sentences that are quite long and difficult to follow. Also, there are a handful of instances where double quotes are used instead of backtics.

**Review by michaelryanpeter — COMMENTED** [review]
Overall, great job! For the most part, my comments are nits/suggestions to improve clarity. 

The only areas that I think need a bit of extra refinement are the feature list and the bug fixes/known issues.

Please feel free to reach out if you have questions or want to run through anything.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (michaelryanpeter) [inline network_observability/network-observability-operator-release-notes.adoc:49]: I know that this hasn't merged yet, but PTAL at the updated guidance for using `{context}` in assemblies: https://github.com/openshift/openshift-docs/pull/71472

**Comment by jotak** [comment]
@skrthomas I'll review it asap (surely tomorrow) but just to not forget: we'll need to add something about [this ticket](https://issues.redhat.com/browse/NETOBSERV-1470) and probably somewhere near the top of the release note, because users may have to manually change a config when they upgrade to get performance improvements...

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:27]: This is a good incongruity to point out. I realize that my other features don't have "In 1.5..." so I think I should just remove this instance of it.

Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:27]: Oops, yes the end of this xref is missing! Copy/paste error.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:31]: eBPF isn't actually spelled out anywhere in the release notes. It is spelled out in the product documentation, but yeah I can see your point its not common to OCP.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:35]: I agree. I reworked this one.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:39]: I think actually this needs to be "In 1.4, these metrics were only available..."

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:53]: I like it. Good suggestion thank you!

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:56]: I think majority are fragments so I'll keep it fragments. I don' think they all need to be complete sentences it would be kinda unnecessarily verbose. But pt taken about cohesion and good detail.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:75]: All the graphs are treated as GUI items in the web console, so it gets bold treatment.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:49]: All my assemblies use {context} so once this merges, I'll need to update them all probably in a separate PR, but thanks for the awareness.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:88]: Got it. Thanks for taking a first pass at a few. I'll see what I can do about the others you mention :)

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [inline network_observability/network-observability-operator-release-notes.adoc:67]: @jotak re: https://github.com/netobserv/network-observability-operator/pull/566

wdyt? Keep it here? Move it elsewhere?

Comment (nathan-weinberg) [reply network_observability/network-observability-operator-release-notes.adoc:67]: I like the placement, one tweak I might make: 
`to enhance eBPF performance when Kafka is being used.`

Comment (jotak) [reply network_observability/network-observability-operator-release-notes.adoc:67]: I think it should be more visible, and clearly call for user action, because users upgrading won't have this set automatically. I would move it as the first thing in notable enhancements, and write something like:
"When upgrading from an existing installation, this new value will not be set automatically in your configuration. If you monitor a performance regression with the eBPF Agent memory consumption after upgrading, you might consider reducing the `kafkaBatchSize` value.

Comment (jotak) [reply network_observability/network-observability-operator-release-notes.adoc:67]: ah sorry I just realize there is an "important" section below, my bad

File: network_observability/network-observability-operator-release-notes.adoc
Comment (jotak) [inline network_observability/network-observability-operator-release-notes.adoc:71]:
```suggestion
This change has no effect when upgrading from an existing installation of Network Observability. In this situation, if you monitor a performance regression with the eBPF Agent memory consumption, you can apply this change manually by reducing the `kafkaBatchSize` value.
```

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc]: I don't think I'll take this. I usually avoid making the Operator possessive because of https://www.ibm.com/docs/en/ibm-style?topic=grammar-possessives, not to make the possessive form of abbreviations or product names.

Comment (michaelryanpeter) [reply network_observability/network-observability-operator-release-notes.adoc]: NP. Do what you feel is best. :) FWIW, I definitely wouldn't use a possessive for <product_name> Operator. I do use "Operator's," as a generic reference, when it is the most direct construction.

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc]: Addressed! TY!

File: network_observability/network-observability-operator-release-notes.adoc
Comment (michaelryanpeter) [inline network_observability/network-observability-operator-release-notes.adoc:89]:
```suggestion
With the `FlowCollector` `v1beta2` API update, you can configure the `spec.loki.readTimeout` specification to update this value according to the Loki Operator `queryTimeout` limit. (link:https://issues.redhat.com/browse/NETOBSERV-1443[*NETOBSERV-1443*])
```

File: network_observability/network-observability-operator-release-notes.adoc
Comment (michaelryanpeter) [inline network_observability/network-observability-operator-release-notes.adoc:91]:
```suggestion
* Previously, the Operator bundle did not display/set some of the supported CSV annotations, such as `features.operators.openshift.io/`.
With this fix/release/update, these annotations are set in the CSV. (link:https://issues.redhat.com/browse/NETOBSERV-1305[*NETOBSERV-1305*])
```

**Review by michaelryanpeter — COMMENTED** [review]
I left a few more suggestions. Looking great!

File: network_observability/network-observability-operator-release-notes.adoc
Comment (skrthomas) [reply network_observability/network-observability-operator-release-notes.adoc:85]: Hmmm...good suggestion. I think I will take some of it and leave some detail too.

**Review by jotak — APPROVED** [review]
thanks @skrthomas !
lgtm

**Pull request merged**
